src/main.rs

```diff
@@ -1,23 +1,115 @@
-use llvm_ir::{Function, Module};
+use std::{collections::HashMap, option::Option::Some};
+
+use llvm_ir::{BasicBlock, Module, Operand, constant::Constant, instruction::Instruction, name::Name::{Name, Number}, operand::Operand::{LocalOperand, ConstantOperand, MetadataOperand}, terminator::Terminator};
+
+struct BFInterp {
+    mem: HashMap<String, u64>,
+}
+
+impl BFInterp {
+
+    fn new() -> BFInterp {
+        BFInterp { 
+            mem: HashMap::new(),
+        }
+    }
+
+    fn interpret(&mut self, module: Module) -> Result<(), &'static str> {  
+        let main_func = match module.get_func_by_name("main") {
+            Some(main) => main,
+            None => return Err("No main function"),
+        };
+
+        match self.it_bb(&main_func.name, &main_func.basic_blocks) {
+            Some(_ret) => return Ok(()),
+            None => return Ok(()),
+        }
+    }
+    
+    fn it_bb(&mut self, func_name: &String, bbs: &Vec<BasicBlock>) -> Option<u64> {
+        for bb in bbs.iter() {
+            for inst in bb.instrs.iter() {
+                match inst {
+                    Instruction::Alloca(_) => {},
```

Comment:
I guess this can't be a `todo` because the test won't run otherwise. I'm not sure yet if we want/need to do anything with `alloca`s. For example, we could use them to verify the IR, but let's leave this empty for now and maybe add a fixme: `FIXME: Do we need to do something with allocas?`.